python/pyarrow/tests/test_array.py

```diff
@@ -1951,6 +1952,13 @@ def test_time32_time64_from_integer():
     assert result.equals(expected)
 
 
+def test_uuid_to_binary():
+    values = [uuid.uuid4() for _ in range(32)]
+    values_bytes = [value.bytes for value in values]
+    arrow_values = pa.array(values)
+    assert arrow_values.to_pylist() == values_bytes
```

Comment:
Ok, but what type is inferred? I would expect `binary(16)`, is it the case?

Reply:
Good catch.  It was `binary()`.  I updated it to `fixed_size_binary` but that raises the question of whether or not we want to support converting heterogeneous arrays of uuid/str/bytes.  At the moment it does so (falling back to `pa.binary`) but only because I don't clear `keep_going` which means it needs to scan the entire array of UUIDs to ensure there is no str/bytes in there first.  So converting homogeneous arrays of UUID would be slightly faster if we didn't support heterogeneous arrays.

Reply:
I don't think performance really is a concern at this point. We should pick the most desirable semantics (I have no idea what those are :-)).